### PR TITLE
#159231776 Filter assets by date

### PR DESCRIPTION
**Pull request by bahdcoder**
**What does this PR do?**

Adds functionality for filtering assets by date added

**Description of Task to be completed?**

 - update datepicker interface with date-change-event interface
 - update tests for filter-dropdown
 - add date-added state to single-category-component
 - add date-added filter algorithm

**How should this be manually tested?**

- visit `macbooks/assets/4`
- click `Filter` button. 
- Add date range for date added.
- Watch assets being filtered.

**Any background context you want to provide?**

xxx

**What are the relevant pivotal tracker stories?**

[#158560264](https://www.pivotaltracker.com/story/show/158560264)

**Screenshots (if appropriate)**
screen shot 2018-07-24 at 8 29 26 pm

**Questions:**

xxx

### CODE REVIEW

File: src/components/DatePicker/interfaces.d.ts [hunk 1]
```diff
@@ -4,12 +4,14 @@ export interface DatePickerState {
...
+}
+
 export interface DatePickerProps {
   id: string;
   date: Moment;
-  onDateChange: (event: {
-    name: string,
-    date: Moment
-  }) => void;
+  onDateChange: (event: DateChangeEvent) => void;
```
Comment (missating) [inline src/components/DatePicker/interfaces.d.ts]: Use `camelcase` for naming rather than `pascal`

Comment (bahdcoder) [reply src/components/DatePicker/interfaces.d.ts]: Interfaces should be named in `pascal`.

**Comment by andela-kakpobome** [comment]
@bahdcoder what's the `-` about in your PR title?

File: src/components/FilterDropdown/index.tsx [hunk 2]
```diff
@@ -15,8 +15,8 @@ import DatePicker from 'components/DatePicker';
 import { FilterDropdownProps } from './interfaces';
 
 const FilterDropdown = (props: FilterDropdownProps) => {
-  const renderDatePicker = name => ((
-    <DatePicker name={name} date={null} onDateChange={() => {}} id={name} />
+  const renderDatePicker = (name, value) => ((
+    <DatePicker name={name} date={value} onDateChange={props.handleDateFieldChange} id={name} />
```
Comment (PaulKariukiRimiru) [inline src/components/FilterDropdown/index.tsx]: Place the `name`,  `date` `onDateChange` and `id` props on separate new lines.

File: src/pages/SingleCategoryPage/index.tsx [hunk 3]
```diff
@@ -84,6 +89,42 @@ export class SingleCategoryPage extends React.Component<SingleCategoryPageProps,
...
+      () => {
+        if (this.state.dateAdded.from && this.state.dateAdded.to) {
+          this.searchAssets();
+        }
+      }
+    );
+  }
+
+  /**
+   * Calls the searchAssets props
```
Comment (PaulKariukiRimiru) [inline src/pages/SingleCategoryPage/index.tsx]: Add new line between the description and first param.

File: src/store/modules/assets/index.ts [hunk 4]
```diff
@@ -1,9 +1,12 @@
 import { GET_ASSETS_FAILURE, GET_ASSETS_SUCCESS } from './types';
 
+import { DateAddedField } from 'pages/SingleCategoryPage/interfaces';
 import { Asset, AssetsActionFailure, AssetsActionSuccess } from './interfaces';
 
 import { assetProps } from './fixtures';
 
+import * as moment from 'moment';
+
```
Comment (PaulKariukiRimiru) [inline src/store/modules/assets/index.ts]: Add import comments.

File: src/pages/SingleCategoryPage/SingleCategoryPage.test.tsx [hunk 5]
```diff
@@ -126,6 +128,33 @@ describe('SingleCategoryPage component', () => {
     expect(singleCategoryPageComp.instance().state.searchTerm).toBe('');
   });
 
+  describe('the handleDateFieldChange function', () => {
+    const wrapper = shallow(<SingleCategoryPage { ...props } />);
+    const { handleDateFieldChange } = wrapper.instance() as any;
+    const now = moment();
+    it('sets dateAdded-from and dateAdded-to when called', () => {
```
Comment (andela-kakpobome) [inline src/pages/SingleCategoryPage/SingleCategoryPage.test.tsx]: Add a line space before this

File: src/pages/SingleCategoryPage/SingleCategoryPage.test.tsx [hunk 6]
```diff
@@ -126,6 +128,33 @@ describe('SingleCategoryPage component', () => {
...
+      handleDateFieldChange({
+        name: 'dateAdded-from',
+        date: now,
+      });
+
+      expect(wrapper.state().dateAdded.from).toEqual(now);
+    });
+    it('searches assets when dateAdded-from and dateAdded-to are provided', () => {
```
Comment (andela-kakpobome) [inline src/pages/SingleCategoryPage/SingleCategoryPage.test.tsx]: Add line spaces between the test cases

File: src/pages/SingleCategoryPage/SingleCategoryPage.test.tsx [hunk 7]
```diff
@@ -126,6 +128,33 @@ describe('SingleCategoryPage component', () => {
...
+    const wrapper = shallow(<SingleCategoryPage { ...props } />);
+    const { handleDateFieldChange } = wrapper.instance() as any;
+    const now = moment();
+    it('sets dateAdded-from and dateAdded-to when called', () => {
+      handleDateFieldChange({
+        name: 'dateAdded-from',
+        date: now,
+      });
+
+      expect(wrapper.state().dateAdded.from).toEqual(now);
```
Comment (andela-kakpobome) [inline src/pages/SingleCategoryPage/SingleCategoryPage.test.tsx]: Also assert that `from` value was not `now` before the call to `handleDateFieldChange`

File: src/pages/SingleCategoryPage/SingleCategoryPage.test.tsx [hunk 8]
```diff
@@ -126,6 +128,33 @@ describe('SingleCategoryPage component', () => {
...
+        dateAdded: {
+          to: now,
+        },
+      });
+      handleDateFieldChange({
+        name: 'dateAdded-from',
+        date: now,
+      });
+
+      expect(props.getAssets).toHaveBeenCalled();
```
Comment (andela-kakpobome) [inline src/pages/SingleCategoryPage/SingleCategoryPage.test.tsx]: Shouldn't you test that it's search assets that's called with the corresponding dates?

File: src/components/FilterDropdown/index.tsx (hunk 2 above)
Comment (andela-kakpobome) [reply src/components/FilterDropdown/index.tsx]: +1

File: src/pages/SingleCategoryPage/index.tsx (hunk 3 above)
Comment (andela-kakpobome) [reply src/pages/SingleCategoryPage/index.tsx]: +1

File: src/store/modules/assets/index.ts (hunk 4 above)
Comment (andela-kakpobome) [reply src/store/modules/assets/index.ts]: +1

File: src/pages/SingleCategoryPage/SingleCategoryPage.test.tsx [hunk 9]
```diff
@@ -126,6 +128,33 @@ describe('SingleCategoryPage component', () => {
...
+    });
+  });
+
```
Comment (andela-kakpobome) [inline src/pages/SingleCategoryPage/SingleCategoryPage.test.tsx]: Also add an integration test to ensure that the right set of data is returned, given the different search criteria

**Comment by andela-kakpobome** [comment]
Why's this PR still not ready, it's been up for too long

**Pull request merged**